src/uct/base/uct_iface.h

```diff
@@ -673,4 +673,26 @@ size_t uct_iov_total_length(const uct_iov_t *iov, size_t iovcnt)
     return total_length;
 }
 
+/**
+ * Copy data to target am_short buffer
+ */
+static UCS_F_ALWAYS_INLINE
+void uct_am_short_fill_data(void *buffer, uint64_t header, const void *payload,
+                            size_t length)
+{
+    /**
+     * Helper structure to fill send buffer of short messages for
+     * non-accelerated transports
+     */
+    struct uct_am_short_packet {
+        uint64_t header;
+        char     payload[];
+    } UCS_S_PACKED *packet = (struct uct_am_short_packet*)buffer;
+
+    packet->header = header;
+    /* suppress false positive diagnostic from uct_self_ep_am_short call */
+    /* cppcheck-suppress ctunullpointer */
```

Comment:
can we comment it in self func only?

Reply:
let's try

Reply:
doesn't work :(

Reply:
@brminich I bit played with __attribute__(nonnull) - it doesn't work, assert doesn't work too :(

Reply:
ok, then - just supress